src/bin/pg_autoctl/state.h

```diff
@@ -50,6 +50,11 @@ typedef enum
 	APPLY_SETTINGS_STATE,
 	PREPARE_MAINTENANCE_STATE,
 	WAIT_MAINTENANCE_STATE,
+	REPORT_LSN_STATE,
```

Comment:
Before running the "election" of the new primary, we absolutely need the current LSN position of every standby node in the group. The fact that the replication is paused has to do with the fact that we don't have a primary anymore. So I see it the way the name of the state implies really.

Reply:
Naming discussions are always hard to finalize :) So, my final suggestion could be `STOP_REPLICATION_AND_REPORT_LSN_STATE` as without reading the code, it'd not be easy for anyone to think that `REPORT_LSN_STATE` also sets the `primary_conninfo` to NULL and stops the replication. 

> The fact that the replication is paused has to do with the fact that we don't have a primary anymore.

Not having primary sounds relative to any node in the cluster. The monitor might deduce it whereas a standby may still be connected to it. In fact, for that case, we're forcefully breaking all connections to the primary, and I feel like being explicit is better in the naming of the state.

That said, if you think strongly about `REPORT_LSN_STATE `, I don't have any further comments here.

Reply:
The reason why we disconnect is to handle network splits. When the primary doesn't have connections from any standby, and also can't connect to the monitor, after 30s it demotes itself (stops Postgres). We need that to avoid split brains in cases of network splits. 

Again, the purpose of this state is actually to have the standby nodes report their current LSN positions. The fact that we shut down the connection to the primary is required for the FSM to proceed *because* the only reason why we ask standby nodes to report their LSN is to proceed with an election and promote one of them.

I like REPORT_LSN better than the alternatives for the moment...